patcher.sh

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env bash
+
+# exit on any error to avoid showing everything was successfull even tho it wasnt
+set -eou pipefail
+
+# Patching user variables
+OSCHOICE=""
+AM2RZIP=""
+HQMUSIC=false
+SYSTEMWIDE=false
+APPIMAGE=false
+PREFIX=""
+
+# Patching internal variables
+# Since people are likely to double click on this, I need a way to get the script_dir
+dirResult="$(dirname '${BASH_SOURCE[0]}')"
+SCRIPT_DIR="$(realpath $dirResult)"
+VERSION="1.5.5"
+OUTPUT="${SCRIPT_DIR}/am2r_${VERSION}"
+RESOURCES="${OUTPUT}/assets"
+
+patch_am2r ()
+{
+	# Set prefix to default value if empty
+	if [ -z "$PREFIX" ]; then
+		if [ "$SYSTEMWIDE" = true ]; then
+			PREFIX="/usr/local"
+		else
+			PREFIX="$SCRIPT_DIR"
+		fi
+	fi
+
+	# We need to do variable adjustments based on the prefix
+	if [ "$SYSTEMWIDE" = true ]; then
+		OUTPUT="${PREFIX}/opt/am2r"
+		RESOURCES="${OUTPUT}/assets"
+	else
+		OUTPUT="${PREFIX}/am2r_${VERSION}"
+		RESOURCES="${OUTPUT}/assets"
+	fi
+	# Cleanup in case the dirs/files exists
+	cleanupDirs=("$OUTPUT" "$SCRIPT_DIR/AM2R.AppDir" "$SCRIPT_DIR/AM2R-x86_64.AppImage" "$SCRIPT_DIR/AM2RWrapper/" "$SCRIPT_DIR/utilities/android/assets/" "$SCRIPT_DIR/AM2RWrapper.apk")
+	rm -rf "${cleanupDirs[@]}"
```

Comment:
This doesn't have to be an array as it's not used later. You can just pass all files to `rm`:
```suggestion
	rm -rf "$OUTPUT" "$SCRIPT_DIR/AM2R.AppDir" "$SCRIPT_DIR/AM2R-x86_64.AppImage" "$SCRIPT_DIR/AM2RWrapper/" "$SCRIPT_DIR/utilities/android/assets/" "$SCRIPT_DIR/AM2RWrapper.apk"
```
The comment should already explain what its doing.

Reply:
Was thinking at first that this'd be less readable, then I remembered we do the exact same stuff multiple times later.
So applied locally it is.